src/Minter.sol

```diff
@@ -85,8 +85,13 @@ contract Minter is IMinter, AccessControlDefaultAdminRules, ReentrancyGuard, Lin
      * @dev Requires the mint configuration not to be cancelled, the user to be whitelisted (if applicable), and not to have minted before (if applicable). Transfers the payment and mints the NFTs.
      * @param configId ID of the mint configuration to use
      * @param merkleProof Proof for whitelist verification, if required
+     * @param maxPrice Maximum price the user is willing to pay
      */
-    function mint(uint256 configId, bytes32[] calldata merkleProof) public payable nonReentrant onlyEOA {
+    function mint(
```

Comment:
This system is not compatible with ETH as a payment token. When using this system you would be sending the maxPrice of ETH and if you pay less the contract should send you back the leftovers. So either we put this refund mech in place our we block variable price mint configs from being created with ETH